docs/adr/0001-create-research-template-docker-repo.md

```diff
@@ -0,0 +1,25 @@
+# 1. Create the research-template-docker repo
+
+Date: 2024-06-04
+
+## Status
+
+Accepted
+
+## Context
+
+We are using [dev containers](https://code.visualstudio.com/docs/devcontainers/create-dev-container) to provide a configuration for Codespaces that gives researchers a useful development environment to use while working on their OpenSAFELY projects. We have created a custom Docker image for this dev container environment. This custom image requires a Dockerfile and other supporting files to be created.
```

Comment:
Nygard states that this section should describe "the forces at play". I don't think the Dockerfile is a force. We could copy it from repository to repository, just as we copy `.devcontainer/devcontainer.json`. However, I think minimising redundancy is a force. Furthermore, as this is the "root" ADR, I think we should expand on why we have decided to use dev containers, Docker images, and Codespaces. What about:

> We know that researchers often struggle to configure their development environments, which degrades their experience of working with OpenSAFELY. In response, we decided to configure a standard development environment using a custom dev container and Docker image, which researchers will interact with in Codespaces.
>
> The code for each OpenSAFELY project lives within a separate repository within the opensafely organisation. Whilst some configuration for the standard development environment _must_ live alongside the code -- because Codespaces are started from repositories -- this results in redundancy. We want to minimise redundancy, and hence to minimise the cost of maintaining the standard development environment.

Reply:
That's a good point. I think the force in this case was the need to pre-build the Docker image. I've updated the text to make that clearer.

We were already using dev containers and Codespaces before we decided to create this repo and even before we started this initiative. It's probably a decision we should record, but I don't think this ADR is the right place.

Reply:
Let me check my understanding...

The architecture decision is "Store `x` in opensafely/research-template and `y` in opensafely-core/research-template-docker", where `x` is something like "dev container configuration" and `y` is something like "Docker image configuration".

We made the architecture decision because it means we can pre-build the Docker image. But why do we want to pre-build the Docker image? The answers to this question are the "forces at play". We want to pre-build the Docker image:

* so the researcher doesn't have to
* to reduce start up time

Reply:
Your understanding is correct. I've added some additional context to explain why we're pre-building the image.